src/mscorlib/shared/System/UnitySerializationHolder.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using System.Reflection;
+
+namespace System
+{
+    /// <summary>
+    /// Holds Null class for which we guarantee that there is only ever one instance of.
+    /// This only exists for backwarts compatibility with 
+    /// </summary>
+#if CORECLR
+    internal
+#else
+    public  // On CoreRT this must be public.
+#endif
+    sealed class UnitySerializationHolder : ISerializable, IObjectReference
+    {
+        internal const int NullUnity = 0x0002;
+        private readonly int _unityType;
+
+        /// <summary>
+        /// A helper method that returns the SerializationInfo that a class utilizing 
+        /// UnitySerializationHelper should return from a call to GetObjectData. It contains
+        /// the unityType (defined above) and any optional data (used only for the reflection types).
+        /// </summary>
+        public static void GetUnitySerializationInfo(SerializationInfo info, int unityType, string data, Assembly assembly)
```

Comment:
`GetUnitySerializationInfo` can be made internal and the "data" and "assembly" parameters serve no purpose anymore.

Aside from that, LGTM.

Reply:
the method itself needs to stay public as otherwise I wouldn't be able to access it from DBNull

Reply:
forget my comment. was wrong.

Reply:
Why wouldn't `DBNull` be able to access it? It lives in the same assembly as `UnitySerializationHolder`.